### PR TITLE
Add /persistent/flags/ to firmware fileio access.

### PR DESCRIPTION
**Pull request by kmdade**

### CODE REVIEW

File: board/piksiv3/rootfs-overlay/etc/init.d/S81sbp_fileio_daemon_firmware [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
-cmd="sbp_fileio_daemon -p 'ipc:///var/run/sockets/fileio_firmware.sub' -s 'ipc:///var/run/sockets/fileio_firmware.pub' --basedir $NDB_DIR --mtd"
+cmd="sbp_fileio_daemon -p 'ipc:///var/run/sockets/fileio_firmware.sub' -s 'ipc:///var/run/sockets/fileio_firmware.pub' --basedir $NDB_DIR --basedir $FLAG_DIR --mtd"
 dir="/"
 user="fileio"
 
 setup_permissions()
 {
   configure_file_resource $user $MTD_FILE 0640
   configure_dir_resource_rec $user $NDB_DIR 0755 0644
+  configure_dir_resource_rec $user $FLAG_DIR 0444
```
Comment (silverjam) [inline board/piksiv3/rootfs-overlay/etc/init.d/S81sbp_fileio_daemon_firmware]: Need 2 sets of perms here, first is directory perms, second is file perms
```suggestion
  configure_dir_resource_rec $user $FLAG_DIR 0555 0444
```

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v2.0.0-develop-2018101616-3-g2fc56f83
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=726641)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.0.0-develop-2018101616-3-g2fc56f83&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.0.0-develop-2018101616-3-g2fc56f83&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v2.0.0-develop-2018101616-3-g2fc56f83
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.0.0-develop-2018101616-3-g2fc56f83)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.0.0-develop-2018101616-3-g2fc56f83/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.0.0-develop-2018101616-3-g2fc56f83

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v2.0.0-develop-2018101616-5-g74c21439
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=726696)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.0.0-develop-2018101616-5-g74c21439&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.0.0-develop-2018101616-5-g74c21439&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v2.0.0-develop-2018101616-5-g74c21439
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.0.0-develop-2018101616-5-g74c21439)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.0.0-develop-2018101616-5-g74c21439/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.0.0-develop-2018101616-5-g74c21439

**Comment by kmdade** [comment]
@silverjam I'm a little stuck on this. I scheduled several more runs and they all came back failing the power-on check. If the fileio daemon gets bad arguments somehow will it tank the whole system at startup?

**Comment by kmdade** [comment]
But the images run fine on my desk. I guess I will try a couple more runs on HITL.

**Comment by silverjam** [comment]
If the fileio daemon isn't running reads from ndb will fail, and (what fails the start up) is reading the device DNA.

**Comment by silverjam** [comment]
Any logs? I can take a look.

**Comment by silverjam** [comment]
Works fine on bench, not sure what the issue is.

**Comment by anonymous-travis-swiftnav** [comment]
## HITL smoke tests: v2.0.0-develop-2018101616-16-g24ec292b
These test runs are kicked off whenever you push a new commit to this PR. All passfail metrics in these runs must pass for the `hitl/pass-fail` status to be marked successful.
### job status
+ [live-roof-650-townsend-post runs](https://gnss-analysis.swiftnav.com/jobs/id=729180)
### gnss-analysis results
At least one run must complete for these links to have data.
#### passfail
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=pass_fail&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.0.0-develop-2018101616-16-g24ec292b&groupby_key=firmware&display_type=table)
#### detailed
+ [live-roof-650-townsend-post](https://gnss-analysis.swiftnav.com/summary_type=q50&metrics_preset=detailed&scenario=live-roof-650-townsend-post&build_type=buildroot_pr&firmware_versions=v2.0.0-develop-2018101616-16-g24ec292b&groupby_key=firmware&display_type=table)

**Comment by anonymous-travis-swiftnav** [comment]
## v2.0.0-develop-2018101616-16-g24ec292b
+ [Artifacts (HITL Dashboard Page)](https://gnss-analysis.swiftnav.com/artifacts/build_type=buildroot_pr&name_filter=v2.0.0-develop-2018101616-16-g24ec292b)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v2.0.0-develop-2018101616-16-g24ec292b/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v2.0.0-develop-2018101616-16-g24ec292b

**Comment by kmdade** [comment]
YAY! merging, thank you @silverjam.

**Pull request merged**